### DISCUSSION TITLE
noob question: how to run your first bot?

### DISCUSSION THREAD

**Discussion by sacrophyte** [Pycord-Development/pycord#1318] [Q&A] [answered]
I am following the guide for "creating your first bot":
https://guide.pycord.dev/getting-started/creating-your-first-bot

Under "Coding the Basics", I appreciate the break down of what the code means, but I don't see where the code is actually run.

When I try to run from my local command line, I get an error:
```
~/discord/bots/my_first_bot$ python3 my_first_bot.py
Python-dotenv could not parse statement starting at line 1
Traceback (most recent call last):
  File "my_first_bot.py", line 6, in <module>
    bot = discord.Bot(debug_guilds=[881207955029110855])
AttributeError: module 'discord' has no attribute 'Bot'
```

It is not clear to me how to actually get the bot running in discord. Would any kind souls mind providing guidance for me, please?

**UPDATE**
I found another example that seems to work better. I see two problems with the pycord guide:

1. I believe the .env file should have one (and only one) line, the TOKEN. The quotes (```) are throwing things off a bit.
2. Is there no bot module in discord? I rewrote it with client, and that seemed to work better.

The pycord guide should still have some mention about actually running the bot. The next section is all about hosting, which is an important discussion, but on this page for "creating your first bot", it is a bit disappointing when following all the instructions does not give you an online, functional bot.

Here is the other article I used for reference:
https://www.freecodecamp.org/news/create-a-discord-bot-with-python/

**Comment by BobDotCom** [comment, marked as answer]
Make sure you have the correct version installed, the current stable version doesn't have `discord.Bot`. Try running `pip uninstall discord discord.py` and `pip install py-cord~=2.0.0b7`, then try again.

**Reply by sacrophyte** [reply]
I'm sorry, but your response is confusing - the example in readme.rst, for version 1.7.3, says "bot = discord.Bot()". Is 1.7.3 the current stable version? If discord.Bot is not in 1.7.3, why is discord.Bot still in the readme.rst as a valid example? Am I reading something wrong?

**Reply by Lulalaby** [reply]
Please help yourself with our official guide: https://guide.pycord.dev

If you have further questions, please join our discord server and ask in the help channel for help. https://discord.gg/pycord 

Our docs are located at https://docs.pycord.dev 

Closing this discussion here now because it's not the right place.

**Reply by Lulalaby** [reply]
In our discord server you can find our updates channel. There you find informations about the correct version btw

**Reply by ItsNeil17** [reply]
@BobDotCom You should edit your message as 2.0 stable as been released & this post is still one of the ones I see at the top of the discussion

**Comment by BatteTarte** [comment]
Here is a simple model :
```python
import discord
from discord.ext import commands

intents = discord.Intents().all()

bot = commands.Bot(intents=intents)

@bot.event
async def on_ready():
    print(f'{bot.user} is now online')
@bot.event
async def on_message(message):
    if message.author == bot.user:
        return
    elif message.content == "ping":
        await message.reply("pong")
```
